### PR TITLE
[driver] Fix prng seed by default

### LINKED ISSUE #2313: [driver][convolutions] The verification cache feature is broken
The reason is highlighted at https://github.com/ROCmSoftwarePlatform/MIOpen/pull/2234#discussion_r1289253848

Let's either remove the feature (`--verification_cache` or `-C`) or revert #2234.

I've collected ~90GB of verification cache files but can survive without it and use GPU verification in _almost all_ cases (except cases when I want to validate the Naive solvers themselves or when HIP compiler is broken). So I prefer reverting #2234 but I am not insisting.

@junliume @JehandadKhan @gonidelis @CAHEK7 what do you think?

----
[Attribution] @junliume @JehandadKhan
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/debugging
- https://github.com/ROCmSoftwarePlatform/MIOpen/labels/urgency_normal
- Proposed assignees:
  - @gonidelis 
  - @CAHEK7 
  - @atamazov

### PR DESCRIPTION
**Pull request by CAHEK7**
Proposed solution for #2313

Probably new env variable should be documented somewhere.

@atamazov could you take a look?

### CODE REVIEW

File: driver/random.hpp [hunk 1]
```diff
@@ -1,11 +1,23 @@
...
 std::minstd_rand& get_minstd_gen()
 {
-    static thread_local std::minstd_rand minstd_gen(std::random_device{}());
+    static thread_local std::minstd_rand minstd_gen{[]() {
+        auto external_seed = miopen::EnvvarValue("MIOPEN_DRIVER_PRNG_SEED", 100500);
+
+        auto seed = external_seed == 0
+                        ? std::random_device{}()
+                        : static_cast<std::random_device::result_type>(external_seed);
+        MIOPEN_LOG_I("Random seed: " << seed);
```
Comment (atamazov) [inline driver/random.hpp]: Please go not use MIOPEN_LOG macros in the driver. Just output to `cout`.

**Review by atamazov — CHANGES_REQUESTED** [review]
Almost good!

What if we'll use environment variable (e.g. `MIOPEN_DRIVER_PRNG_LEGACY=1/0` etc) to fall back to the old code? Developers who want to use old cached data (like me) would appreciate that. I suspect that regeneration of huge amount of data may take dozen of hours.

**Comment by CAHEK7** [comment]
Added MIOPEN_USE_LEGACY_PRNG and MIOPEN_DRIVER_PRNG_SEED env variables.
If MIOPEN_DRIVER_PRNG_SEED == 0, it means random seed - can be useful for smoke testing.

**Comment by dmikushin** [comment]
Please take note that std::random_device [takes some costs](https://stackoverflow.com/a/39301751/4063520) and side-effects associated with random state creation taken seriously. Almost definitely it should be enough to produce a time()-based seed.

**Pull request closed without merging**